### PR TITLE
issue #495: Add a dashboard

### LINKED ISSUE #495: [UX] Add a useful dashboard to backdrop
It would be useful for backdrop to include a useful administrative dashboard. 

Remaining items (before merge)
- [x] Add icon for dashboard to admin bar
- [x] Change all instances of 'administer' to 'manage'
- [x] Remove "Powered by backdrop" footer block
---

Ideas summarized from comments (can be done before or after merge):

- [x] Add icons to blocks (match those in admin bar)
- [x] Change human names for left/right regions (in Sutro layout)
- [x] Add an available updates block (access limited)
- [ ] Copy dashboard permissions into the settings form
- [ ] Add an unpublished (incl. scheduled) content block
- [ ] Add new blocks with more information

---
~~PR by @docwilmot: https://github.com/backdrop/backdrop/pull/2400~~
~~PR by @docwilmot: https://github.com/backdrop/backdrop/pull/2452~~
~~PR by @jenlampton: https://github.com/backdrop/backdrop/pull/2524~~
~~PR by @quicksketch: https://github.com/backdrop/backdrop/pull/2632~~
PR by @jenlampton: https://github.com/backdrop/backdrop/pull/2633

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/495

### CODE REVIEW

**Comment by jenlampton** [comment]
Initial feedback on PR: **I LOVE IT!**  It's already a huge improvement over "no dashboard" :)

Lots of minor nits:

1)  We should remove the text `Dashboard admin -` from all the block titles that display on the dashboard itself. These make sense in the `Add block` dialog, but not after the blocks have been placed. 

screen shot 2018-12-24 at 2 28 02 pm

2) the same goes for the word "Dashboard" in the following block titles: `Dashboard create content`, `Dashboard overview - Content & User accounts`, `Dashboard overview - Content`. Great on the `Add block` screen, but overkill as the block title.

3) As per backdrop norm, we shouldn't refer to Accounts as "Users" but the full `User accounts` wherever possible.

screen shot 2018-12-24 at 2 31 06 pm

5) To match the rest of the admin interface, we should use `Manage` instead of `Administer`. 

screen shot 2018-12-24 at 2 37 43 pm

6a) The "Dashboard welcome" can probably just be a custom block (not reusable).

6b) The block text should be updated to mention "blocks" instead of "content panes". `Welcome to your dashboard. Edit this layout to add more blocks here, or configure those provided by default.`

6c) The description text shouldn't mention "Total Control" :)

7) I'm not sure we need the `Dashboard overview - Content & User accounts` block at all. The two separate blocks are probably sufficient, especially if we place them both on the dashboard initially.

8) I'm unable to place the `Dashboard overview - User accounts` block. I get a spinner when I click "Add block", but the block won't appear. There are some related PHP notices in the logs.

9) All the block descriptions currently start with "Displays" or "provides" but neither word is necessary. For scan-ability it would be better if we removed those words and got straight to the point. 

screen shot 2018-12-24 at 2 46 02 pm

10a) On the settings page, we still refer to the permission as the `Have Total Control permission` but that should probably be something like 'Access dashboard` and live in system.module ...

10b) and this setting should probably live in system.core - Which may require an update hook?)

screen shot 2018-12-24 at 2 49 51 pm

11) the description there should probably be more definite. Maybe something like `Everyone with the "Access dashboard" permission will be automatically redirected to the Dashboard when they log in.` 

12) The dashboard might feel less cluttered if we moved the links at the bottom of each admin block from the left to the right. 
screen shot 2018-12-24 at 2 58 51 pm

File: core/modules/layout/includes/dashboard_blocks/block.create.inc [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<?php
```
Comment (jenlampton) [inline core/modules/layout/includes/dashboard_blocks/block.create.inc:1]: Can we move this block to node module?

File: core/modules/layout/includes/dashboard_blocks/block.menus.inc [hunk 2]
```diff
@@ -0,0 +1,144 @@
+<?php
```
Comment (jenlampton) [inline core/modules/layout/includes/dashboard_blocks/block.menus.inc:1]: Can we move this block to menu module?

File: core/modules/layout/includes/dashboard_blocks/block.node_types.inc [hunk 3]
```diff
@@ -0,0 +1,143 @@
+<?php
```
Comment (jenlampton) [inline core/modules/layout/includes/dashboard_blocks/block.node_types.inc:1]: Move to node module?

File: core/modules/layout/includes/dashboard_blocks/block.overview_content.inc [hunk 4]
```diff
@@ -0,0 +1,69 @@
+<?php
```
Comment (jenlampton) [inline core/modules/layout/includes/dashboard_blocks/block.overview_content.inc:1]: Move to Node module?

File: core/modules/layout/includes/dashboard_blocks/block.overview_user.inc [hunk 5]
```diff
@@ -0,0 +1,77 @@
+<?php
```
Comment (jenlampton) [inline core/modules/layout/includes/dashboard_blocks/block.overview_user.inc:1]: Move to user module?

File: core/modules/layout/includes/dashboard_blocks/block.taxonomy.inc [hunk 6]
```diff
@@ -0,0 +1,161 @@
+<?php
```
Comment (jenlampton) [inline core/modules/layout/includes/dashboard_blocks/block.taxonomy.inc:1]: Move to taxonomy module?

File: core/modules/layout/layout.admin.inc [hunk 7]
```diff
@@ -2884,3 +2884,22 @@ function layout_get_block_usage($module = NULL, $delta = NULL) {
   }
   return $blocks;
 }
+
+/**
+ * menu callback for settings page
+ */
+function layout_dashboard_admin_settings() {
```
Comment (jenlampton) [inline core/modules/layout/layout.admin.inc:2891]: Move to dashboard.module?

File: core/modules/layout/layout.module [hunk 8]
```diff
@@ -326,6 +328,28 @@ function layout_menu() {
     'file' => 'layout.admin.inc',
   );
 
+
+  // Add dashboard and its settings menu if the layout is enabled.
```
Comment (jenlampton) [inline core/modules/layout/layout.module:332]: Move to dashboard module.

File: core/modules/layout/layout.module [hunk 9]
```diff
@@ -514,6 +538,18 @@ function layout_theme() {
     'layout_region_inner' => array(
       'variables' => array('blocks' => array(), 'tag' => '', 'classes' => array(), 'attributes' => array()),
     ) + $base,
+    'dashboard_create' => array(
```
Comment (jenlampton) [inline core/modules/layout/layout.module]: this permission can be moved to dashboard module.

Comment (jenlampton) [inline core/modules/layout/layout.module:541]: hm...  not sure where these should go?

File: core/modules/layout/layout.module [hunk 10]
```diff
@@ -573,6 +609,11 @@ function layout_permission() {
       'description' => t('Add and modify layouts on pages.'),
       'restrict access' => TRUE,
     ),
+    'access dashboard' => array(
```
Comment (jenlampton) [inline core/modules/layout/layout.module:612]: Move to dashboard module.

File: core/modules/layout/layout.module [hunk 11]
```diff
@@ -594,6 +635,14 @@ function layout_config_info() {
     'label_key' => 'path',
     'group' => t('Layout menu items'),
   );
+  $prefixes['layout.layout.dashboard'] = array(
```
Comment (jenlampton) [inline core/modules/layout/layout.module:638]: Move to Dashboard module.

**Comment by herbdool** [comment]
I'm still hoping this will get a comment block before it gets merged. Something simple like "There are X comments ready for approval." Hopefully not difficult to add.

**Comment by jenlampton** [comment]
> I'm still hoping this will get a comment block before it gets merged.

That's something we may be able to add between now and the 15th, too.

**Pull request closed without merging**